src/azure-cli-core/azure/cli/core/tests/test_generic_update.py

```diff
@@ -39,69 +70,71 @@ def my_set(**kwargs): #pylint:disable=unused-argument
+
+        # Test simplest ways of setting properties
+        app.execute('update-obj --set myProp=newValue'.split())
+        self.assertEqual(my_obj.my_prop, 'newValue', 'set simple property')
+        
+        app.execute('update-obj --set myProp=val3'.split())
+        self.assertEqual(my_obj.my_prop, 'val3', 'set simple property again')
 
-        app.execute('gencommand --set prop=val2'.split())
-        self.assertEqual(my_obj['prop'], 'val2', 'set simple property')
+        app.execute('update-obj --set myList[0]=newValA'.split())
+        self.assertEqual(my_obj.my_list[0], 'newValA', 'set simple list element')
 
-        app.execute('gencommand --set prop=val3'.split())
-        self.assertEqual(my_obj['prop'], 'val3', 'set simple property again')
+        app.execute('update-obj --set myDict.myCamelKey=success'.split())
+        self.assertEqual(my_obj.my_dict['myCamelKey'], 'success', 'set simple dict element with camel case key')
 
-        app.execute('gencommand --set list[0]=f'.split())
-        self.assertEqual(my_obj['list'][0], 'f', 'set simple list element')
+        app.execute('update-obj --set myDict.my_snake_key=success'.split())
+        self.assertEqual(my_obj.my_dict['my_snake_key'], 'success', 'set simple dict element with snake case key')
 
-        app.execute('gencommand --set list[2][0]=g'.split())
-        self.assertEqual(my_obj['list'][2][0], 'g', 'set nested list element')
+        # Test the different ways of indexing into a list of objects or dictionaries by filter
+        app.execute('update-obj --set myListOfCamelDicts[myKey=value_2].myKey=new_value'.split())
+        self.assertEqual(my_obj.my_list_of_camel_dicts[1]['myKey'], 'new_value', 'index into list of dictionaries by camel-case key')
 
-        app.execute('gencommand --set list[2][1].d=h'.split())
-        self.assertEqual(my_obj['list'][2][1]['d'], 'h', 'set nested dict element')
+        app.execute('update-obj --set myListOfSnakeDicts[my_key=value2].my_key=new_value'.split())
+        self.assertEqual(my_obj.my_list_of_snake_dicts[1]['my_key'], 'new_value', 'index into list of dictionaries by snake-case key')
 
-        app.execute('gencommand --set list[0]={} list[0].foo=bar'.split())
-        self.assertEqual(my_obj['list'][0]['foo'], 'bar', 'replace nested scalar with new dict')
+        app.execute('update-obj --set myListOfObjects[myString=1.2.3.4].myString=new_value'.split())
+        self.assertEqual(my_obj.my_list_of_objects[1].my_string, 'new_value', 'index into list of objects by key')
 
-        app.execute('gencommand --set list[1]=[] --add list[1] key1=value1'
-                    ' --set list[1][0].key2=value2'.split())
-        self.assertEqual(my_obj['list'][1][0]['key1'], 'value1',
-                         'replace nested scalar with new list with one value')
-        self.assertEqual(my_obj['list'][1][0]['key2'], 'value2',
-                         'add a second value to the new list')
```

Comment:
I added replacing an entry with a list, adding a value, and then setting a second value in a separate command (lines 62-67) 

---
In reply to: [79196616](https://github.com/Azure/azure-cli/pull/867#discussion_r79196616) [](ancestors = 79196616,78653673)